packages/jira-adapter/src/filters/assets/assets_object_type_path.ts

```diff
@@ -0,0 +1,63 @@
+/*
+*                      Copyright 2023 Salto Labs Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+import { InstanceElement, isInstanceElement } from '@salto-io/adapter-api'
+import { collections } from '@salto-io/lowerdash'
+import { DAG } from '@salto-io/dag'
+import { pathNaclCase } from '@salto-io/adapter-utils'
+import { FilterCreator } from '../../filter'
+
+const { awu } = collections.asynciterable
+
+const createPaths = async (assetsObjectTypes: InstanceElement[]): Promise<void> => {
+  const graph = new DAG<InstanceElement>()
+  assetsObjectTypes.forEach(assetsObjectType => {
+    const parentFullName = assetsObjectType.value.parentObjectTypeId?.value.elemID.name
+    const dependencies = parentFullName ? [parentFullName] : []
+    graph.addNode(
+      assetsObjectType.elemID.name,
+      dependencies,
+      assetsObjectType,
+    )
+  })
+  await awu(graph.evaluationOrder()).forEach(
+    graphNode => {
+      const instance = graph.getData(graphNode.toString())
+      const parentPath = instance.value.parentObjectTypeId?.value.path
+      if (parentPath === undefined) {
+        instance.path = [...instance.path?.slice(0, -1) ?? [],
+          pathNaclCase(instance.value.name), pathNaclCase(instance.value.name)]
```

Comment:
some questions - 
1. why do we have `pathNaclCase(instance.value.name)` twice?
2.  if there is no `parentObjectTypeId` shouldn't we leave the path as it is?

maybe a comment can be helpful here :)

+ readability nit:
```suggestion
        instance.path = [
        ...instance.path?.slice(0, -1) ?? [],
          pathNaclCase(instance.value.name),
          pathNaclCase(instance.value.name)
        ]
```

Reply:
Answers:
1. I want each file to be inside its folder. 
2. No, because than the files are not in a folder and they can have subFolders inside them.